File: 프로그래머스/0/120882. 등수 매기기/등수 매기기.swift
import Foundation

func solution(_ score:[[Int]]) -> [Int] {
    var dic: [Int:Int] = [:]
    let average = score.map { $0.reduce(0, +) }
    var cnt = 1
    
    average.sorted(by: >).forEach {
        if !dic.keys.contains($0) { dic.updateValue(cnt, forKey: $0) }
        cnt += 1
    }
    
    return average.map { dic[$0]! }
}

File: 프로그래머스/0/120882. 등수 매기기/등수 매기기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<vector<int>> score) {
    vector<int> answer;
    vector<double> avg;

    for (vector<int> v : score) {
        avg.push_back((v[0] + v[1]) / 2.0);
    }

    for (int i = 0; i < avg.size(); i++) {
        double val = avg[i];
        int cnt = 1;
        for (int j = 0; j < avg.size(); j++) {
            if(val < avg[j]) {
                cnt++;
            }
        }
        answer.push_back(cnt);
    }

    return answer;
}